torch/csrc/distributed/rpc/types.cpp

```diff
@@ -0,0 +1,54 @@
+#include <torch/csrc/distributed/rpc/types.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+GloballyUniqueId::GloballyUniqueId(worker_id_t createdOn, local_id_t localId)
+    : createdOn_(createdOn), localId_(localId) {}
+
+bool GloballyUniqueId::operator==(const GloballyUniqueId& other) const {
+  return createdOn_ == other.createdOn_ && localId_ == other.localId_;
+}
+
+bool GloballyUniqueId::operator!=(const GloballyUniqueId& other) const {
+  return createdOn_ != other.createdOn_ || localId_ != other.localId_;
+}
+
+at::IValue GloballyUniqueId::toIValue() const {
+  std::vector<at::IValue> ivalues = {(int64_t)createdOn_, (int64_t)localId_};
+  return c10::ivalue::Tuple::create(std::move(ivalues));
+}
+
+GloballyUniqueId GloballyUniqueId::fromIValue(const at::IValue& ivalue) {
+  auto ivalues = ivalue.toTuple()->elements();
+  TORCH_CHECK(
+      ivalues.size() == 2,
+      "Constructing GloballyUniqueId from ivalue "
+      "expects a GenericList of two elements, but got ",
+      ivalues.size());
+
+  worker_id_t createdOn = ivalues[0].toInt();
+  local_id_t localId = ivalues[1].toInt();
+
+  TORCH_CHECK(
+      createdOn < std::numeric_limits<worker_id_t>::max(),
+      "GloballyUniqueId createdOn out of range, got ",
+      createdOn);
+
+  TORCH_CHECK(
+      localId < std::numeric_limits<local_id_t>::max(),
```

Comment:
The localId actually has a lower max since it can use only 48 bits.

Reply:
In this PR `GloballyUniqueId` is a struct of two numbers, and the 48 bits is only applied when calculating hash. We can add those when we consolidate this with autograd context id.